File: HW/HW2/hw2.py
class Solution:
    def firstBadVersion(self, n):
        left, right = 1, n
        while left < right:
            mid = left + (right - left) // 2
            if isBadVersion(mid):
                right = mid
            else:
                left = mid + 1
        return left

    def isBadVersion(self, version):
        return version >= 5  # Placeholder value for testing.

File: HW/HW2/hw2.java
public class Solution {
    public int firstBadVersion(int n) {
        int left = 1, right = n;
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (isBadVersion(mid)) {
                right = mid;
            } else {
                left = mid + 1;
            }
        }
        return left;
    }

    // This is a placeholder for the isBadVersion method.
    // You should replace it with the actual implementation provided by the API.
    private boolean isBadVersion(int version) {
        // Your implementation here.
        return false;
    }
}

